implementations/rust/ockam/ockam_command/src/http/inlet/create.rs

```diff
@@ -0,0 +1,566 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use clap::builder::NonEmptyStringValueParser;
+use clap::Args;
+use colorful::Colorful;
+use miette::{miette, IntoDiagnostic};
+use ockam_api::nodes::service::tcp_outlets::Outlets;
+use tracing::trace;
+use url::Url;
+
+use crate::node::util::initialize_default_node;
+use crate::shared_args::OptionalTimeoutArg;
+use crate::tcp::util::alias_parser;
+use crate::{Command, CommandGlobalOpts, Error};
+use ockam::identity::Identifier;
+use ockam::transport::HostnamePort;
+use ockam::Context;
+use ockam_abac::PolicyExpression;
+use ockam_api::address::extract_address_value;
+use ockam_api::cli_state::journeys::{
+    JourneyEvent, NODE_NAME, TCP_INLET_ALIAS, TCP_INLET_AT, TCP_INLET_CONNECTION_STATUS,
+    TCP_INLET_FROM, TCP_INLET_TO,
+};
+use ockam_api::cli_state::{random_name, CliState};
+use ockam_api::colors::color_primary;
+use ockam_api::nodes::models::portal::InletStatus;
+use ockam_api::nodes::service::tcp_inlets::Inlets;
+use ockam_api::nodes::BackgroundNodeClient;
+use ockam_api::{fmt_info, fmt_log, fmt_ok, fmt_warn, ConnectionStatus};
+use ockam_core::api::{Reply, Status};
+use ockam_multiaddr::proto::{self, Secure};
+use ockam_multiaddr::{MultiAddr, Protocol as _};
+use ockam_node::compat::asynchronous::resolve_peer;
+
+use crate::util::parsers::duration_parser;
+use crate::util::parsers::hostname_parser;
+use crate::util::{find_available_port, port_is_free_guard, process_nodes_multiaddr};
+
+/// Create TCP Inlets
+#[derive(Clone, Debug, Args)]
+pub struct CreateCommand {
```

Comment:
You should be able to ""inherit"" fields by declaring a field containing the inlet `Create` command with the `#[command(flatten)]` attribute. From the clap point of view would be the same as-if the fields were locally declared.